src/core/credentials/transport/tls/spiffe_utils.cc

```diff
@@ -111,19 +120,16 @@ absl::Status validatePath(absl::string_view path) {
 
 }  // namespace
 
-// spiffe://trust-domain-name/path
 absl::StatusOr<SpiffeId> SpiffeId::FromString(absl::string_view uri) {
-  if (absl::Status status = doInitialUriValidation(uri); !status.ok()) {
-    return status;
-  }
-  if (!absl::StartsWithIgnoreCase(uri, SPIFFE_PREFIX)) {
+  GRPC_RETURN_IF_ERROR(DoInitialUriValidation(uri));
+  if (!absl::StartsWithIgnoreCase(uri, kSpiffePrefix)) {
```

Comment:
Why are we ignoring the case here?  I don't see anything in the SPIFFE spec that says we should accept other cases for the scheme.